### PR TITLE
API Improvement and Refactoring

### PR DESCRIPTION
**Pull request by StevenCostiou**
Plus fixing some critiques along the way.

### CODE REVIEW

File: src/Sindarin/SindarinDebugger.class.st [hunk 1]
```diff
@@ -70,14 +70,13 @@ SindarinDebugger >> contextIsAboutToSignalException: aContext [
...
-	node isMessage ifFalse: [ ^ false ].
+	self isMessageSend ifFalse: [ ^ false ].
 	(#( #signal #signalIn: ) includes: node selector) ifFalse: [ ^ false ].
 	aContext basicSize >= 1 ifFalse: [ ^ false ].
 	(Exception allSubclasses includes: (aContext at: aContext basicSize))
 		ifTrue: [ ^ true ]. "#signal sent to a subclass of Exception"
-	(Exception allSubclasses includes:
-		 (aContext at: aContext basicSize) class) ifTrue: [ ^ true ]. "#signal sent to an instance of a subclass of Exception"
-	^ false
+	^(Exception allSubclasses includes: (aContext at: aContext basicSize) class) "#signal sent to an instance of a subclass of Exception"
```
Comment (adri09070) [inline src/Sindarin/SindarinDebugger.class.st:78]: Cannot it be simplified with `isKindOf:` ?

It's still ugly but ...

File: src/Sindarin/SindarinDebugger.class.st [hunk 2]
```diff
@@ -144,9 +143,7 @@ SindarinDebugger >> jumpIntoBlock: aBlockNode toNode: targetNode [
 SindarinDebugger >> message: aSelector [
 	"Returns whether the execution is about to send a message of selector @aSelector to any object"
 
-	[ ^ self messageSelector = aSelector ]
-		on: Error
-		do: [ ^ false ]
+	^ self isMessageSend and: [ self messageSelector = aSelector ]
```
Comment (adri09070) [inline src/Sindarin/SindarinDebugger.class.st:146]: Ok, but the method name is weird, I don't expect it to return a boolean.

It should be renamed: `isMessageSend:`, or something like that

File: src/Sindarin/SindarinDebugger.class.st (hunk 1 above)
Comment (StevenCostiou) [reply src/Sindarin/SindarinDebugger.class.st:78]: Perhaps.
It is ugly and it should be simplified.
This is also part of the original code.

File: src/Sindarin/SindarinDebugger.class.st [hunk 3]
```diff
@@ -25,17 +25,17 @@ Class {
 { #category : #stackAccessHelpers }
 SindarinDebugger >> assignmentValue [
 	"Returns the value about to be assigned, if the current node is an assignment node. Otherwise, returns nil"
-
-	self node isAssignment ifFalse: [
+	self flag: 'Why there is no error raised here, while for the case of message sends there is an error?'.
```
[the remark this answers is not in the document]
Comment (StevenCostiou) [reply src/Sindarin/SindarinDebugger.class.st:28]: OK it did not take my comment when I clicked "resolve conversation".
I wrote that I was just flagging odd stuff for further work once this PR is integrated.

File: src/Sindarin/SindarinDebugger.class.st (hunk 2 above)
Comment (StevenCostiou) [reply src/Sindarin/SindarinDebugger.class.st:146]: Same my comment disappeared.
I said this is part of the original code and a deeper refactoring should be done in a new PR.

**Review by adri09070 — APPROVED** [review]

**Pull request merged**